### ISSUE TITLE
[RNN/LSTM]concatenation error during invoke()

### ISSUE THREAD

**Issue by Raj-Gohil** [tensorflow/tensorflow#57988]
### 1. System information

- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux
- TensorFlow installation (pip package or built from source): pip
- TensorFlow library (version, if pip package or github SHA, if built from source):

### 2. Code

I converted pytorch model to tensorflow using onnx. The output of converted tensorflow model is exactly as expected.
The following is the pytorch model code:

        def __init__(self, device: Union[str, torch.device]=None, verbose=True, weights_fpath: Union[Path, str]=None):
        super().__init__()
        self.lstm = nn.LSTM(mel_n_channels, model_hidden_size, model_num_layers, batch_first=True)
        self.linear = nn.Linear(model_hidden_size, model_embedding_size)
        self.relu = nn.ReLU()
        if device is None:
            device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
        elif isinstance(device, str):
            device = torch.device(device)
        self.device = device

        # Load the pretrained model'speaker weights
        if weights_fpath is None:
            weights_fpath = Path(__file__).resolve().parent.joinpath("pretrained.pt")
        else:
            weights_fpath = Path(weights_fpath)

        if not weights_fpath.exists():
            raise Exception("Couldn't find the voice encoder pretrained model at %s." %
                            weights_fpath)
        start = timer()
        checkpoint = torch.load(weights_fpath, map_location="cpu")
        self.load_state_dict(checkpoint["model_state"], strict=False)
        self.to(device)

        if verbose:
            print("Loaded the voice encoder model on %s in %.2f seconds." %
                  (device.type, timer() - start))

    def forward(self, mels: torch.FloatTensor):
        _, (hidden, _) = self.lstm(mels)
        embeds_raw = self.relu(self.linear(hidden[-1]))
        return embeds_raw / torch.norm(embeds_raw, dim=1, keepdim=True)

I then converted the tesorflow .pb model to tensorflow lite using the code:

```
import tensorflow as tf
converter = tf.lite.TFLiteConverter.from_saved_model("test_new1")
converter.optimizations = [tf.lite.Optimize.DEFAULT]
converter.experimental_new_converter=True
converter.target_spec.supported_ops = [tf.lite.OpsSet.TFLITE_BUILTINS,tf.lite.OpsSet.SELECT_TF_OPS]
tfmodel = converter.convert()
open('test_new1/saved_lite.tflite', 'wb').write(tfmodel)
```

I then tried to infer using the tflite model using the following code:

```
import tensorflow as tf
interpreter = tf.lite.Interpreter(r'test_new1/saved_lite.tflite')
input_details = interpreter.get_input_details()
output_details= interpreter.get_output_details()
interpreter.set_tensor(input_details[0]['index'],dummy_input)
interpreter.invoke()
```

The output of the input_details and output_details is:
```
[{'name': 'serving_default_actual_input_1:0',
  'index': 0,
  'shape': array([  1, 160,  40], dtype=int32),
  'shape_signature': array([ -1, 160,  40], dtype=int32),
  'dtype': numpy.float32,
  'quantization': (0.0, 0),
  'quantization_parameters': {'scales': array([], dtype=float32),
   'zero_points': array([], dtype=int32),
   'quantized_dimension': 0},
  'sparsity_parameters': {}}]
[{'name': 'StatefulPartitionedCall:0',
  'index': 154,
  'shape': array([  1, 256], dtype=int32),
  'shape_signature': array([ -1, 256], dtype=int32),
  'dtype': numpy.float32,
  'quantization': (0.0, 0),
  'quantization_parameters': {'scales': array([], dtype=float32),
   'zero_points': array([], dtype=int32),
   'quantized_dimension': 0},
  'sparsity_parameters': {}}]
```
  
 The size of the dummy_input is [1,160,40]
 
 The error I am getting is:
` tensorflow/lite/kernels/concatenation.cc:158 t->dims->data[d] != t0->dims->data[d] (296 != 1)Node number 3 (CONCATENATION) failed to prepare.Node number 48 (WHILE) failed to invoke.`
 
 
Can anyone help me out with the this issue as I am converting a pretrained pytorch model to tflite. The issue I am facing is during the conversion of tensorflow to tflite because till tensorflow it is perfectly working fine.

**Comment by mohantym** [comment]
Hi @Raj-Gohil! 
Could you confirm that input shape for both original model and lite model is [1,160,40] . If not same , you can resize the input tensor in lite model to shape of original model. 

Would it possible to share a toy model/ Colab gist too.

Thank you!

**Comment by Raj-Gohil** [comment]
Hello @mohantym,
The input shape is [1,160,40]
If you want i can share the .pb file.

**Comment by mohantym** [comment]
Ok @Raj-Gohil !
Did you check the fusion code lab instructions ? Please share the model file too.
Thank you!

**Comment by Raj-Gohil** [comment]
[test_new1.zip](https://github.com/tensorflow/tensorflow/files/9724024/test_new1.zip)
This is the folder which has the .pb and .tflite model.
Please check it out and I did not get what exactly is to be done according to the fushion code lab instructions.
If you can share a code snippet of what needs to be done in python... it would be great.

**Comment by Raj-Gohil** [comment]
Do you have updates on this @mohantym

**Comment by mohantym** [comment]
@Raj-Gohil 
I was able to reproduce concatentation.cc error with above method. Proposed idea on Fusion code lab was like below.
( But it was throwing dimension mismatch)

```
run_model = tf.function(lambda x: model(x)) 
BATCH_SIZE = 1
STEPS = 160
INPUT_SIZE = 40
concrete_func = run_model.get_concrete_function(
    tf.TensorSpec([BATCH_SIZE, STEPS, INPUT_SIZE], model.inputs[0].dtype))

# model directory.
MODEL_DIR = "keras_lstm"
model.save(MODEL_DIR, save_format="tf", signatures=concrete_func)

converter = tf.lite.TFLiteConverter.from_saved_model(MODEL_DIR)
tflite_model = converter.convert()
```

Attached [gist](https://colab.sandbox.google.com/gist/mohantym/49fd8ddd795dbd5a08dbcc4b71e36a15/git_57988.ipynb#scrollTo=YE5BjPHfCT1y) for reference.

Thank you!

**Comment by Raj-Gohil** [comment]
Hello @sachinprasadhs .... do we have any update on this?

**Comment by pjpratik** [comment]
Hi @Raj-Gohil 

As this [comment](https://github.com/tensorflow/tensorflow/issues/56804#issuecomment-1292708824) suggests, the issue can be while authoring the model. Can you please check the input shape compatibility with the channels while converting to tensorflow?

Thanks.
